pkg/renderer/sgml/html5/icon.go

```diff
@@ -0,0 +1,33 @@
+package html5
+
+const (
+	// Inline icons are presented as the icon as an image, or font icon, or the text "[class]" where class
+	// is the Alt text, and defaults to the icon type.  Unlike asciidoctor, we do place class settings on
+	// the image to match rotate, flip, and size, allowing images to be manipulated with css style just
+	// like the icons can.  We also allow for an ID to be placed on the icon, as another enhancement.
+	inlineIconTmpl = `<span {{ if .ID }}id="{{ .ID }}" {{ end}}class="icon{{ if .Role }} {{ .Role }}{{ end }}">` +
+		`{{ if .Link }}<a class="image" href="{{ .Link }}"{{ if .Window }} target="{{ .Window }}"{{ end }}>{{ end }}` +
+		`{{ .Icon }}` +
+		`{{ if .Link }}</a>{{ end }}` +
+		`</span>`
+
+	iconImageTmpl = `<img src="{{ .Path }}"` +
+		`{{ if .Alt }} alt="{{ .Alt }}{{ end }}"` +
```

Comment:
it's a good idea to use the raw strings to format the template! It makes it more readable 👍

Reply:
Yeah, with this many if-blocks it was the only way I could do this and retain my sanity.